.github/workflows/ci-post-land.yml

```diff
@@ -0,0 +1,351 @@
+name: ci-post-land
+
+on:
+  create:
+    branches: [main, release-*]
+  push:
+    branches: [main, release-*] # "gha-test-*"
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  max_threads: 16
+  pre_command: cd /opt/git/diem/
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      changes-target-branch: ${{ steps.changes.outputs.changes-target-branch }}
+      changes-base-git-rev: ${{ steps.changes.outputs.changes-base-git-rev }}
+      changes-base-githash: ${{ steps.changes.outputs.changes-base-githash }}
+      changes-pull-request-number: ${{ steps.changes.outputs.changes-pull-request-number }}
+      rust-changes: ${{ steps.rust-changes.outputs.changes-found }}
+      base-image-changes: ${{ steps.base-image-changes.outputs.changes-found }}
+      update-osx-sccache: ${{ steps.environment.outputs.name }}
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          # This ensures that the tip of the PR is checked out instead of the merge between the base ref and the tip
+          # On `push` this value will be empty and will "do-the-right-thing"
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0 #get all the history!!!
+      - name: Git Hooks and Checks
+        run: ./scripts/git-checks.sh
+      - id: changes
+        name: determine changes
+        uses: diem/actions/changes@faadd16607b77dfa2231a8f366883e01717b3225
+        with:
+          workflow-file: ci-post-land.yml
+          github-token: ${{secrets.GITHUB_TOKEN}}
+      - id: rust-changes
+        name: find rust/cargo changes.
+        uses: diem/actions/matches@faadd16607b77dfa2231a8f366883e01717b3225
+        with:
+          pattern: '^documentation\|^docker\|^scripts\|^developers.diem.com'
+          invert: "true"
+      - id: base-image-changes
+        name: find dev-setup.sh/base docker image changes
+        uses: diem/actions/matches@faadd16607b77dfa2231a8f366883e01717b3225
+        with:
+          pattern: '.github/workflows/ci-post-land.yml\|.github/actions/dockerhub_login/action.yml\|docker/ci/github/Dockerfile\|scripts/dev_setup.sh\|rust-toolchain'
+
+  update-sccache-osx:
+    environment:
+      name: Sccache
+    runs-on: macos-11
+    if: ${{ needs.prepare.outputs.rust-changes == 'true' && github.event_name == 'push' }}
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          # This ensures that the tip of the PR is checked out instead of the merge between the base ref and the tip
+          # On `push` this value will be empty and will "do-the-right-thing"
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0 #get all the history!!!
```

Comment:
is the full history needed here (and further down, after prepare step) ?

Reply:
for any git comparison work it's needed, the prepare step does this, rust steps do this when running tests (a git comparison).    The default pull from github actions has no history.    The pull's fast enough there's almost no reason not to be consistent everywhere even if a job doesn't need history.

Reply:
Ah -  " rust steps do this when running tests (a git comparison)".  That explains it. Thanks !